### ISSUE TITLE
UX Enhancement - File Browser

### ISSUE THREAD

**Issue by tgeorgeux** [jupyterlab/jupyterlab#6616]
### Uploading a folder
I was testing the upload functionality and it doesn't look like you're able to upload an entire folder. Is this intentional behavior?

I was using jakevdp's iPython Handbook as an example and it was pretty tedious to upload all the elements and remake the folder structure in the JupyterLab file browser.

**Comment by jasongrout** [comment]
No, uploading a folder isn't implemented, and probably would be hard with browsers.

**Comment by vidartf** [comment]
Multi-file upload, where you select all the files in a folder might be possible though.

**Comment by tgeorgeux** [comment]
Multi-file upload is already functioning, this is simply a place to wax philosophical about the possibility of uploading folders.

**Comment by vidartf** [comment]
Hmm, there is a draft attribute that is decently supported: https://caniuse.com/#feat=input-file-directory
